mapit_global/management/commands/mapit_global_import.py

```diff
@@ -236,14 +236,13 @@ def verbose(s):
 
                 area_type = Type.objects.get(code=type_directory)
 
-                try:
-                    osm_code = Code.objects.get(type=code_type_osm,
-                                                code=osm_id,
-                                                area__generation_low__lte=current_generation,
-                                                area__generation_high__gte=current_generation)
-                except Code.DoesNotExist:
+                osm_code = Code.objects.filter(type=code_type_osm,
+                                               code=osm_id,
+                                               area__generation_low__lte=current_generation,
+                                               area__generation_high__gte=current_generation
+                                               ).order_by('-area_id').first()
```

Comment:
I think this is surprising enough that it'd be worth adding a comment along the lines of the explanation in the pull request as a comment in the code too.